lib/connection/connection_config.d.ts

```diff
@@ -0,0 +1,170 @@
+     * The login name for your Snowflake user or your Identity Provider (e.g. your login name for Okta).
+     */
+    username?: string;
+
+    /**
+     * Password for the user.
+     */
+    password?: string;
+
+    /**
+     * Specifies the authenticator to use for verifying user login credentials.
+     */
+    authenticator?: string;
+
+    /**
+     * Specifies the hostname of an authenticated proxy server.
+     */
+    proxyHost?: string;
+
+    /**
+     * Specifies the password for the user specified by proxyUser.
+     */
+    proxyPort?: number;
+
+
+    serviceName?: string;
+
+    /**
+     * Specifies the private key (in PEM format) for key pair authentication.
+     */
+    privateKey?: string;
+
+    /**
+     * Specifies the local path to the private key file (e.g. rsa_key.p8)
+     */
+    privateKeyPath?: string;
+
+    /**
+     * Specifies the passcode to decrypt the private key file, if the file is encrypted.
+     */
+    privateKeyPass?: string;
+
+    /**
+     * Specifies the OAuth token to use for authentication. Set this option if you set the authenticator option to OAUTH.
+     */
+    token?: string;
+
+    /**
+     * The default virtual warehouse to use for the session after connecting. Used for performing queries, loading data, etc.
+     */
+    warehouse?: string;
+
+    /**
+     * The default database to use for the session after connecting.
+     */
+    database?: string;
+
+    /**
+     * The default schema to use for the session after connecting.
+     */
+    schema?: string;
+
+    /**
+     * The default security role to use for the session after connecting.
+     */
+    role?: string;
+
+    /**
+     * Specifies how to return results that contain duplicate column names.
+     */
+    rowMode?: RowMode;
+
+    /**
+     * Enabling this parameter causes the method to return a Node.js Readable stream, which you can use to consume rows as they are received. 
+     */
+    streamResult?: boolean;
+
+    /**
+     * return the following data types as strings: Boolean, Number, Date, Buffer, and JSON.
+     */
+    fetchAsString?: DataType[] | undefined;
+
+    /**
+     * By default, client connections typically time out approximately 3-4 hours after the most recent query was executed.
+     */
+    clientSessionKeepAlive?: boolean;
+
+    /**
+     * Sets the frequency (interval in seconds) between heartbeat messages.
+     */
+    clientSessionKeepAliveHeartbeatFrequency?: number;
+
+    /**
+     * To convert Snowflake INTEGER columns to JavaScript Bigint, which can store larger values than JavaScript Number
+     */
+    jsTreatIntegerAsBigInt?: boolean;
+
+    /**
+     * Sets the maximum number of binds the driver uses in a bulk insert operation. The default value is 100000 (100K).
+     */
+    arrayBindingThreshold?: number;
+
+    /**
+     * Validate default options set by users.
+     */
+    validateDefaultParameters?: boolean;
+
+    /**
+    * 
+    */
+    forceStageBindError?: number;
```

Comment:
Please add documentation here

Reply:
I removed this option from the d.ts file because this option seems a testing purpose option.